zhaquirks/tuya/valve.py

```diff
@@ -0,0 +1,200 @@
+                "temperature_change",
+                self.TEMPERATURE_ATTRS[attrid],
+                value * 10,  # decidegree to centidegree
+            )
+        elif attrid == SITERWELL_MODE_ATTR:
+            self.endpoint.device.thermostat_bus.listener_event("mode_change", value)
+        elif attrid == SITERWELL_VALVE_STATE_ATTR:
+            self.endpoint.device.thermostat_bus.listener_event("valve_state", value)
+
+
+class SiterwellGS361Thermostat(LocalDataCluster, Thermostat):
+    """Thermostat cluster for Siterwell GS361 device."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.switch_bus.add_listener(self)
+
+    def temperature_change(self, attr, value):
+        """Local or target temperature change."""
+        self._update_attribute(self.attridx[attr], value)
+
+    def mode_change(self, value):
+        """System Mode change."""
+        if value == 0:
+            mode = self.SystemMode.Off
+        elif value == 1:
+            mode = self.SystemMode.Auto
+        elif value == 2:
+            mode = self.SystemMode.Heat
+        self._update_attribute(self.attridx["system_mode"], mode)
+
+    def valve_state(self, value):
+        """Valve state update."""
+        if value == 0:
+            state = self.RunningMode.Off
+        else:
+            state = self.RunningMode.Heat
+        self._update_attribute(self.attridx["running_mode"], state)
+
+    async def write_attributes(self, attributes, manufacturer=None):
+        """Implement mandatory writeable arguments."""
+
+        records = self._write_attr_records(attributes)
+
+        manufacturer_attrs = {}
+        for record in records:
+            if record.attrid == self.attridx["system_mode"]:
+                if record.value.value == self.SystemMode.Off:
+                    mode = 0
+                elif record.value.value == self.SystemMode.Auto:
+                    mode = 1
+                elif record.value.value == self.SystemMode.Heat:
+                    mode = 2
+                else:
+                    self.error("Unsupported value for SystemMode")
+                    continue
+                manufacturer_attrs[SITERWELL_MODE_ATTR] = mode
+            elif record.attrid == self.attridx["occupied_heating_setpoint"]:
+                manufacturer_attrs[SITERWELL_TEMPERATURE_ATTR] = round(
+                    record.value.value / 10  # centidegrees to decidegrees
+                )
+            elif record.attrid == self.attridx["ctrl_seqe_of_oper"]:
+                # log a wrong value, but otherwise do nothing
+                if record.value.value != self.ControlSequenceOfOperation.Heating_Only:
+                    self.error("Unsupported value for ControlSequenceOfOperation")
+
+        await self.endpoint.tuya_manufacturer.write_attributes(
+            {SITERWELL_MODE_ATTR: mode}, manufacturer=manufacturer
```

Comment:
I think is not {SITERWELL_MODE_ATTR: mode} but manufacturer_attrs

Reply:
Indeed. This got lost in a refactoring I did.